### PR TITLE
Creating postman-collection types

### PR DESCRIPTION
**Pull request by kbuzby**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by plantain-00 — APPROVED** [review]

File: types/postman-collection/index.d.ts [hunk 1]
```diff
@@ -174,25 +176,21 @@ export interface CollectionDefinition extends ItemGroupDefinition {
...
     };
-    item?: Array<ItemDefinition | ItemGroupDefinition>;
     variable?: VariableDefinition;
-    auth?: RequestAuthDefinition;
-    event?: EventDefinition[];
-    version?: string | Version;
 }
 
-export class Collection extends ItemGroup<Request> implements CollectionDefinition {
+export class Collection extends ItemGroup<Request> {
```
Comment (ThePlenkov) [inline types/postman-collection/index.d.ts]: Hi!  I just found a mistake here. It must be ItemGroup<Item> here, not <Request>. Could you please check this? Thank you!

Comment (ThePlenkov) [reply types/postman-collection/index.d.ts]: i tried to use code like this but tslint told me it must be Request. If I put request - i always get default request values in my item at the end.

```
    let item = new Item({
      request: { url: url.toString(), method: init.method }
    });
    this.items.append(item);
```

**Review by ThePlenkov — COMMENTED** [review]
Hi! Could you please review my comments? I tried to render postman collection and tslint was giving me error before I fixed the type.

**Comment by kbuzby** [comment]
This pull request is from nearly 2 years ago, and already merged into master. I think this is better served as a new issue / new PR to fix.
  
  
   Also it's worth mentioning that since this was 2 years ago, and I haven't kept the types up to date, it's possible that newer versions of postman-collection aren't compatible in the same way.
  
  
   
    On February 15, 2020 7:44 AM Petr Plenkov <notifications@github.com> wrote:
   
   
    
   
   
    
   
   @ThePlenkov commented on this pull request.
   Hi! Could you please review my comments? I tried to render postman collection and tslint was giving me error before I fixed the type.
   
   In types/postman-collection/index.d.ts:
   >  }
 
-export class Collection extends ItemGroup<Request> implements CollectionDefinition {
+export class Collection extends ItemGroup<Request> {

   Hi! I just found a mistake here. It must be ItemGroup here, not . Could you please check this? Thank you!
   
   In types/postman-collection/index.d.ts:
   >  }
 
-export class Collection extends ItemGroup<Request> implements CollectionDefinition {
+export class Collection extends ItemGroup<Request> {

   i tried to use code like this but tslint told me it must be Request. If I put request - i always get default request values in my item at the end.
       let item = new Item({
      request: { url: url.toString(), method: init.method }
    });
    this.items.append(item);

**Comment by ThePlenkov** [comment]
@kbuzby Thank you for replying. Do you suggest to raise issue in postman-collection project? Thank you

**Comment by kbuzby** [comment]
If the issue is just with the types I would imagine you'd submit here on DefinitelyTyped - the Postman team isn't responsible for the types. But since this is largely volunteer contributions I'm not sure how quickly it'd get picked up, if at all. Personally, I don't have much free time to make the full pass required to update these to the current version of postman-collection. The best thing would be if you could make the update / fix yourself and submit the PR here!
  
  
   
    On February 15, 2020 3:49 PM Petr Plenkov <notifications@github.com> wrote:
   
   
    
   
   
    
   
   @kbuzby Thank you for replying. Do you suggest to raise issue in postman-collection project? Thank you

**Pull request merged**